tests/test_bases.py

```diff
@@ -22,6 +22,63 @@ def test_get_bases():
     np.testing.assert_array_equal(bases, [[1, 2], [5, 6]])
 
 
+def test_get_bases_no_bases():
```

Comment:
This and subsequent tests can be all done in a single test function or by using [parametrize](https://docs.pytest.org/en/7.1.x/how-to/parametrize.html).

Ok to leave it like this for this PR though